### PR TITLE
Added more detail validation for settings and category

### PR DESCRIPTION
**Pull request by ericc1103**

### CODE REVIEW

File: test/jsonValidationTest.js [hunk 1]
```diff
@@ -5,120 +5,167 @@ const fs = require('fs');
...
 
-    it('Verifying .cohort files', function(done){
+    it('Verifying .cohort files', function (done) {
         let failedList = [];
         browseDirectory(cohortPath, (error, results) => {
             results.filter(file => file.substr(-7) === '.cohort')
-                .forEach(file => {                    
-                    validateJsonString(file)
+                .forEach(file => {
+                    validateJsonStringAndGetObject(file)
```
Comment (gardnerjr) [inline test/jsonValidationTest.js]: I *think* you can go all "javascript" on this block, and just
```
.forEach(validateJsonStringAndGetObject);
```
if you're just passing the param directly to the method

File: test/jsonValidationTest.js [hunk 2]
```diff
@@ -5,120 +5,167 @@ const fs = require('fs');
...
     });
 
-    it('Verifying cohort category files', function(done){
+    it('Verifying cohort category files', function (done) {
         browseDirectory(cohortPath, (error, results) => {
             if (error) throw error;
             results.filter(file => file.substr(-22) === 'categoryResources.json')
                 .forEach(file => {
-                    validateJsonString(file);
+                    let json = validateJsonStringAndGetObject(file);
```
Comment (gardnerjr) [inline test/jsonValidationTest.js]: nit: it isn't "json" after this call, it's an actual object.

File: test/jsonValidationTest.js (hunk 1 above)
Comment (ericc1103) [reply test/jsonValidationTest.js]: @gardnerjr I am not sure what you mean by that.. Could you elaborate more?

File: test/jsonValidationTest.js [hunk 3]
```diff
@@ -5,120 +5,167 @@ const fs = require('fs');
...
+function validateJsonStringAndGetObject(file) {
+    let json = fs.readFileSync(file, 'utf8');
+    let obj = TryParseJson(json)
+    assert.isTrue(obj != null, 'Invalid json format with \'' + file + '\'');
+    return obj;
+}
 
+function validateSettingsForCohort(settings) {
+    if (!settings.hasOwnProperty('$schema')) {
+        assert.fail("The $schema field is missing with '" + file + "'");
```
Comment (gardnerjr) [inline test/jsonValidationTest.js]: this block is repeated like 100 times?

```
function validateSetting(obj: any, name: string, file: string) {
    if (!obj.hasOwnProperty(name)) {
        assert.fail( file + ":  property " + name + " is missing" );
    }
}
```

then use it like:
```
["name", "$schema", "author"].forEach( field => validateSetting(settings, field, file) );
```

in each test you just change what named fields are validated

File: test/jsonValidationTest.js (hunk 2 above)
Comment (ericc1103) [reply test/jsonValidationTest.js]: Done

File: test/jsonValidationTest.js [hunk 4]
```diff
@@ -5,120 +5,167 @@ const fs = require('fs');
...
-      })();
+        })();
     });
-  };
+};
 
+function validateJsonStringAndGetObject(file) {
+    let json = fs.readFileSync(file, 'utf8');
+    let obj = TryParseJson(json)
+    assert.isTrue(obj != null, 'Invalid json format with \'' + file + '\'');
```
Comment (gardnerjr) [inline test/jsonValidationTest.js]: why not catch the error and actually output it in the assert?  if the file is invalid, why not explain why right here?

File: test/jsonValidationTest.js (hunk 1 above)
Comment (gardnerjr) [reply test/jsonValidationTest.js]: i believe you can replace this block:

```.forEach(file => { 
validateJsonStringAndGetObject(file);
});
```

with `.forEach(validateJsonStringAndGetObject);`

File: test/jsonValidationTest.js (hunk 4 above)
Comment (ericc1103) [reply test/jsonValidationTest.js]: Good suggestion

File: test/jsonValidationTest.js (hunk 1 above)
Comment (ericc1103) [reply test/jsonValidationTest.js]: I will just leave as is since it is not significant and leave it for more readability. :-)

**Pull request merged**